synchronizer/synchronizer.go

```diff
@@ -410,6 +396,23 @@ func (s *ClientSynchronizer) Sync() error {
 			}
 			metrics.FullSyncIterationTime(time.Since(start))
 			log.Info("L1 state fully synchronized")
+			// Sync trusted state
+			// latestSyncedBatch -> Last batch on DB
+			// latestSequencedBatchNumber -> last batch on SMC
+			if latestSyncedBatch >= latestSequencedBatchNumber {
+				startTrusted := time.Now()
+				if s.syncTrustedStateExecutor != nil && !s.isTrustedSequencer {
+					log.Info("Syncing trusted state (permissionless)")
+					err = s.syncTrustedState(latestSyncedBatch)
+					metrics.FullTrustedSyncTime(time.Since(startTrusted))
+					if err != nil {
+						log.Warn("error syncing trusted state. Error: ", err)
+						s.CleanTrustedState()
+						continue
+					}
+				}
+				waitDuration = s.cfg.SyncInterval.Duration
+			}
```

Comment:
This code can't be here. This could cause false positive in trusted reorg

Reply:
Let's talk about that. I moved because, if not, start synchronizing from L2 when there are data pending on L1